### PR TITLE
Move renderer phenomenon out from common by introducing different …

### PR DESCRIPTION
**Pull request by jansav**
…implementations for environments

Signed-off-by: Janne Savolainen <janne.savolainen@live.fi>

### CODE REVIEW

File: src/common/utils/is-allowed-resource.injectable.ts [hunk 1]
```diff
@@ -4,16 +4,16 @@
...
+import { allowedResourcesInjectionToken } from "../cluster-store/allowed-resources-injection-token";
 
 export type IsAllowedResource = (resource: KubeResource) => boolean;
 
 const isAllowedResourceInjectable = getInjectable({
   id: "is-allowed-resource",
 
   instantiate: (di, resourceName: string) => {
-    const allowedResources = di.inject(allowedResourcesInjectable);
+    const allowedResources = di.inject(allowedResourcesInjectionToken);
```
Comment (jansav) [inline src/common/utils/is-allowed-resource.injectable.ts:16]: Important bit here. Switching to use `injectionToken`

File: src/main/create-cluster/allowed-resources.injectable.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import { computed } from "mobx";
+import { allowedResourcesInjectionToken } from "../../common/cluster-store/allowed-resources-injection-token";
+
+// TODO: Figure out implementation for this later.
```
Comment (jansav) [inline src/main/create-cluster/allowed-resources.injectable.ts:9]: Important bit here: Dummy implementation for main till we have real one.

**Comment by jansav** [comment]
Everything else is just relocation of files from `common` to `renderer`.

File: src/main/create-cluster/allowed-resources.injectable.ts (hunk 2 above)
Comment (Nokel81) [reply src/main/create-cluster/allowed-resources.injectable.ts:9]: I don't think we can ever have a real one. What I think is that we should try and move `isAllowedResource` to be only within `renderer`.

**Review by Nokel81 — DISMISSED** [review]

File: src/main/create-cluster/allowed-resources.injectable.ts (hunk 2 above)
Comment (jansav) [reply src/main/create-cluster/allowed-resources.injectable.ts:9]: As we discussed in huddle, we surely can. When we have removed the iframe, we can call e.g. `navigateToPods(clusterId)` which then will have the requirement of checking if cluster has pods. Or even application menu containing all clusters and ways to navigate to pages in them. But that's enough speculation :) We'll get back on this when the time comes.

**Comment by jansav** [comment]
Accidently merged: https://github.com/lensapp/lens/pull/5636 to this. Not gonna unmerge them again, since both of these were approved. (just to save time, since these are just chores) @Nokel81 @jim-docker could you re-approve this?

This PR's original changes are in the first commit, so if you need to check it still, review commit by commit.

**Pull request merged**